include/sliding_window/search/do_parallel.hpp

```diff
@@ -4,14 +4,16 @@
 #include <future>
 #include <vector>
 
+#include <seqan3/core/debug_stream.hpp>
+
 namespace sliding_window
 {
 
 template <typename t>
 inline void do_parallel(t && worker, size_t const num_records, size_t const threads, double & compute_time)
 {
     auto start = std::chrono::high_resolution_clock::now();
-    std::vector<decltype(std::async(std::launch::async, worker, size_t{}, size_t{}))> tasks;
+    std::vector<decltype(std::async(std::launch::async, worker, size_t{}, size_t{}))> tasks; // std::async returns a std::future
```

Comment:
Hi @marehr please ignore the rest of the PR for now (it's messy). 

But my question would be about this parallelization that is already there for my application. I take a set of query reads assign them to threads and execute in parallel. I execute the `worker` method which finds corresponding bins to each read. Is this also something that should be taken care of the `slotted_cart_queue`? 

Or would we want the `slotted_cart_queue` to only be used for the next step i.e bulk execution of the STELLAR read mapping?